plugin/src/main/java/org/opensearch/ml/action/agents/GetAgentTransportAction.java

```diff
@@ -0,0 +1,97 @@
+import org.opensearch.ml.common.transport.agent.MLAgentGetRequest;
+import org.opensearch.ml.common.transport.agent.MLAgentGetResponse;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+import lombok.AccessLevel;
+import lombok.experimental.FieldDefaults;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+public class GetAgentTransportAction extends HandledTransportAction<ActionRequest, MLAgentGetResponse> {
+
+    Client client;
+    NamedXContentRegistry xContentRegistry;
+
+    @Inject
+    public GetAgentTransportAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        Client client,
+        NamedXContentRegistry xContentRegistry
+    ) {
+        super(MLAgentGetAction.NAME, transportService, actionFilters, MLAgentGetRequest::new);
+        this.client = client;
+        this.xContentRegistry = xContentRegistry;
+    }
+
+    @Override
+    protected void doExecute(Task task, ActionRequest request, ActionListener<MLAgentGetResponse> actionListener) {
+        MLAgentGetRequest mlAgentGetRequest = MLAgentGetRequest.fromActionRequest(request);
+        String agentId = mlAgentGetRequest.getAgentId();
+        GetRequest getRequest = new GetRequest(ML_AGENT_INDEX).id(agentId);
+
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            client.get(getRequest, ActionListener.runBefore(ActionListener.wrap(r -> {
+                log.debug("Completed Get Agent Request, id:{}", agentId);
+
+                if (r != null && r.isExists()) {
+                    try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, r.getSourceAsBytesRef())) {
+                        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                        MLAgent mlAgent = MLAgent.parse(parser);
+                        actionListener.onResponse(MLAgentGetResponse.builder().mlAgent(mlAgent).build());
+                    } catch (Exception e) {
+                        log.error("Failed to parse ml agent" + r.getId(), e);
+                        actionListener.onFailure(e);
+                    }
+                } else {
+                    actionListener
+                        .onFailure(
+                            new OpenSearchStatusException(
+                                "Failed to find agent with the provided agent id: " + agentId,
+                                RestStatus.NOT_FOUND
+                            )
+                        );
+                }
+            }, e -> {
+                if (e instanceof IndexNotFoundException) {
+                    log.error("Failed to get agent index", e);
+                    actionListener.onFailure(new OpenSearchStatusException("Failed to find agent", RestStatus.NOT_FOUND));
```

Comment:
These two doesn't quite match? What we don't get agent index or agent itself?

Reply:
changed to `Failed to get agent index` to log and throw the same error message. I also changed the tests too.